.github/workflows/docker.yml

```diff
@@ -25,6 +30,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: "true"
+          ref: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.commit || '' }}
```

Comment:
nit: I think `github.event.inputs.commit` is empty if the action is not manually started. so the condition might not be necessary.